repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/contract/ContractPreProcessor.java

```diff
@@ -0,0 +1,300 @@
+import org.apache.atlas.model.instance.EntityMutations;
+import org.apache.atlas.repository.graphdb.AtlasGraph;
+import org.apache.atlas.repository.graphdb.AtlasVertex;
+import org.apache.atlas.repository.store.graph.AtlasEntityStore;
+import org.apache.atlas.repository.store.graph.v2.*;
+import org.apache.atlas.type.AtlasEntityType;
+import org.apache.atlas.type.AtlasTypeRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.*;
+
+import static org.apache.atlas.AtlasErrorCode.*;
+import static org.apache.atlas.repository.Constants.*;
+import static org.apache.atlas.repository.util.AtlasEntityUtils.mapOf;
+import static org.apache.atlas.type.AtlasTypeUtil.getAtlasObjectId;
+
+public class ContractPreProcessor extends AbstractContractPreProcessor {
+    private static final Logger LOG = LoggerFactory.getLogger(ContractPreProcessor.class);
+    public static final String ATTR_ASSET_GUID = "dataContractAssetGuid";
+    public static final String REL_ATTR_LATEST_CONTRACT = "dataContractLatest";
+    public static final String REL_ATTR_GOVERNED_ASSET_CERTIFIED = "dataContractLatestCertified";
+    public static final String REL_ATTR_PREVIOUS_VERSION = "dataContractPreviousVersion";
+    public static final String ASSET_ATTR_HAS_CONTRACT = "hasContract";
+    public static final String ASSET_ATTR_DESCRIPTION = "description";
+
+    public static final String CONTRACT_QUALIFIED_NAME_SUFFIX = "contract";
+    public static final String VERSION_PREFIX = "version";
+    public static final String CONTRACT_ATTR_STATUS = "status";
+    private final AtlasEntityStore entityStore;
+    private final boolean storeDifferentialAudits;
+    private final EntityDiscoveryService discovery;
+
+
+
+    public ContractPreProcessor(AtlasGraph graph, AtlasTypeRegistry typeRegistry,
+                                EntityGraphRetriever entityRetriever, AtlasEntityStore entityStore,
+                                boolean storeDifferentialAudits, EntityDiscoveryService discovery) {
+
+        super(graph, typeRegistry, entityRetriever);
+        this.storeDifferentialAudits = storeDifferentialAudits;
+        this.entityStore = entityStore;
+        this.discovery = discovery;
+    }
+
+    @Override
+    public void processAttributes(AtlasStruct entityStruct, EntityMutationContext context, EntityMutations.EntityOperation operation) throws AtlasBaseException {
+        AtlasEntity entity = (AtlasEntity) entityStruct;
+        switch (operation) {
+            case CREATE:
+                processCreateContract(entity, context);
+                break;
+            case UPDATE:
+                // Updating an existing version of the contract
+                processUpdateContract(entity, context);
+        }
+
+    }
+
+    private void processUpdateContract(AtlasEntity entity, EntityMutationContext context) throws AtlasBaseException {
```

Comment:
shall we prune critical relations here? any relationshipAttribute passed in payload will be catered & relations can be messed up

when pruning, remove complete relationshipAttribute itself instead setting to null or empty arraylist

Reply:
Make sense. Is it true for create as well?

Reply:
Yeah, got it.